File: 0705-design-hashset/0705-design-hashset.java
class MyHashSet {
    List<Integer>[] chain;
    int size = 10000;

    public MyHashSet() {
        chain = new ArrayList[size];
        for(int i=0; i < size; i++){
            chain[i] = new ArrayList<>();
        }    
    }
    
    public void add(int key) {
        int hash = key % size;
        if(!chain[hash].contains(key)){
            chain[hash].add(key);
        }
    }
    
    public void remove(int key) {
        int hash = key % size;
        chain[hash].remove(Integer.valueOf(key));
    }
    
    public boolean contains(int key) {
        int hash = key % size;
        return chain[hash].contains(key);
    }
}

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet obj = new MyHashSet();
 * obj.add(key);
 * obj.remove(key);
 * boolean param_3 = obj.contains(key);
 */

File: 0705-design-hashset/0705-design-hashset.py
class MyHashSet:

    def __init__(self):
        self.size = 10000
        self.chain = [[] for _ in range(self.size)]

    def add(self, key: int) -> None:
        hash = key % self.size

        if key not in self.chain[hash]:
            self.chain[hash].append(key)

    def remove(self, key: int) -> None:
        hash = key % self.size

        if key in self.chain[hash]:
            self.chain[hash].remove(key)

    def contains(self, key: int) -> bool:
        hash = key % self.size
        return key in self.chain[hash]